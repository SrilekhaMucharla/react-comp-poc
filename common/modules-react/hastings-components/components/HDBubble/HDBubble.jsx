import React from 'react';
import PropTypes from 'prop-types';

const HDBubble = ({ children, position }) => (
    <div className="hd-bubble" data-position={position}>
        <span className="hd-bubble__content">
            {children}
        </span>
    </div>
);

HDBubble.propTypes = {
    children: PropTypes.node,
    position: PropTypes.oneOf(['top'])
};

HDBubble.defaultProps = {
    position: 'top',
    children: undefined
};

export default HDBubble;
