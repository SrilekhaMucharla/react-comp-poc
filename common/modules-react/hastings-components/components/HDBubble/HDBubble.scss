$arrow-side: 12px;

.hd-bubble {
    padding: 8px 15px;
    color: $mono-white;
    background-color: $brand-hastings-red;
    border-radius: 2px;
    box-shadow: 0 5px 5px 0 rgba(0, 0, 0, 0.15);
    position: absolute;
    z-index: 1;
    text-align: center;
    right: 50%;
    transform: translateX(50%);
    font-size: 12px;
    line-height: 1.2;
    width: fit-content;
    transition: .2s bottom;

    &__content {
        vertical-align: middle;
    }

    &::after {
        content: '';
        position: absolute;
        background-color: $brand-hastings-red;
        width: $arrow-side;
        height: $arrow-side;
        left: 50%;
        bottom: -($arrow-side / 2);
        transform: translateX(-50%) rotate(45deg);
    }

    @include media-breakpoint-down(md) {
        right: 12%;
        left: 12%;
        width: auto;
    }

    &[data-position="top"] {
        bottom: 100%;
        transform: translateX(50%) translateY(-9px);

        @include media-breakpoint-down(md) {
            transform: translateY(-9px);
        }
    }
}
