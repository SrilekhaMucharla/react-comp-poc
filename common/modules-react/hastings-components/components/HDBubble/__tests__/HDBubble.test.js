import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDBubble from '../HDBubble';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDBubble />', () => {
    it('render component', () => {
        const wrapper = shallow(<HDBubble>Text</HDBubble>);
        expect(wrapper).toMatchSnapshot();
    });
});
