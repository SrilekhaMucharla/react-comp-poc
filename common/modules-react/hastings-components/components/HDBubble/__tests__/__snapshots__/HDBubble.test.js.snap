// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDBubble /> render component 1`] = `
<div
  className="hd-bubble"
  data-position="top"
>
  <span
    className="hd-bubble__content"
  >
    Text
  </span>
</div>
`;
